infra/metrics/main.tf

```
@@ -402,6 +414,108 @@ resource "google_cloudfunctions2_function" "gbfs_validation_report_conversion_ba
+    available_cpu                    = local.function_preprocessed_analytics_config.available_cpu
+    max_instance_request_concurrency = local.function_preprocessed_analytics_config.max_instance_request_concurrency
+    max_instance_count               = local.function_preprocessed_analytics_config.max_instance_count
+    min_instance_count               = local.function_preprocessed_analytics_config.min_instance_count
+    service_account_email            = google_service_account.metrics_service_account.email
+    ingress_settings                 = "ALLOW_ALL"
+    vpc_connector                    = data.google_vpc_access_connector.vpc_connector.id
+    vpc_connector_egress_settings    = "PRIVATE_RANGES_ONLY"
+    dynamic "secret_environment_variables" {
+      for_each     = local.function_preprocessed_analytics_config.secret_environment_variables
+      content {
+        key        = secret_environment_variables.value["key"]
+        project_id = var.project_id
+        secret     = "${upper(var.environment)}_${secret_environment_variables.value["key"]}"
+        version    = "latest"
+      }
+    }
+  }
+
+}
+
+# 2.7 GBFS - Preprocessed analytics function
+resource "google_storage_bucket" "gbfs_analytics_bucket" {
+  location = var.gcp_region
+  name     = "mobilitydata-gbfs-analytics-${var.environment}"
+}
+resource "google_cloudfunctions2_function" "gbfs_preprocessed_analytics" {
+  name        = "${local.function_preprocessed_analytics_config.name}-gbfs"
+  project     = var.project_id
+  description = local.function_preprocessed_analytics_config.description
+  location    = var.gcp_region
+  depends_on  = [google_secret_manager_secret_iam_member.secret_iam_member]
+
+  build_config {
+    runtime     = var.python_runtime
+    entry_point = "${local.function_preprocessed_analytics_config.entry_point}_gbfs"
+    source {
+      storage_source {
+        bucket = google_storage_bucket.functions_bucket.name
+        object = google_storage_bucket_object.function_preprocessed_analytics.name
+      }
+    }
+  }
+  service_config {
+    environment_variables = {
+      PYTHONNODEBUGRANGES = 0
+        ANALYTICS_BUCKET    = google_storage_bucket.gbfs_analytics_bucket.name
+    }
+    available_memory                 = local.function_preprocessed_analytics_config.memory
+    timeout_seconds                  = local.function_preprocessed_analytics_config.timeout
+    available_cpu                    = local.function_preprocessed_analytics_config.available_cpu
+    max_instance_request_concurrency = local.function_preprocessed_analytics_config.max_instance_request_concurrency
+    max_instance_count               = local.function_preprocessed_analytics_config.max_instance_count
+    min_instance_count               = local.function_preprocessed_analytics_config.min_instance_count
+    service_account_email            = google_service_account.metrics_service_account.email
+    ingress_settings                 = "ALLOW_ALL"
```

Should we restrict this function to the following? We can create a follow-up ticket for this, as I can see other functions with this _flexible_ rule.
```suggestion
    ingress_settings                 = "ALLOW_INTERNAL_AND_GCLB"
```

---

Opened #715 to address